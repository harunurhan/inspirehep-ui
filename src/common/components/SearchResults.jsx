import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';

class SearchResults extends Component {
  render() {
    const { renderItem, results } = this.props;
    return (
      <div>
        {results.map(result => (
          <div className="mv3" key={result.get('id')}>
            {renderItem(result)}
          </div>
        ))}
      </div>
    );
  }
}

SearchResults.propTypes = {
  results: PropTypes.instanceOf(Immutable.List),
  renderItem: PropTypes.func.isRequired,
};

SearchResults.defaultProps = {
  results: Immutable.List(),
};

export default SearchResults;
