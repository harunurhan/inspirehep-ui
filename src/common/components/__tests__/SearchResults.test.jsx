import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import SearchResults from '../SearchResults';

describe('SearchResults', () => {
  it('renders with all props set', () => {
    const results = fromJS([
      {
        id: 1,
        value: 'value1',
      },
      {
        id: 2,
        value: 'value2',
      },
    ]);
    const wrapper = shallow(
      <SearchResults
        results={results}
        renderItem={result => <span>{result.get('value')}</span>}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
