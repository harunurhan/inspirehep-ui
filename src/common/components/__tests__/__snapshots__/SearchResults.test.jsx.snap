// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders with all props set 1`] = `
<div>
  <div
    className="mv3"
    key="1"
  >
    <span>
      value1
    </span>
  </div>
  <div
    className="mv3"
    key="2"
  >
    <span>
      value2
    </span>
  </div>
</div>
`;
